lang/src/org/partiql/lang/eval/EvaluatingCompiler.kt

```diff
@@ -180,8 +179,8 @@ internal class EvaluatingCompiler(
      * Base class for [ExprAggregator] instances which accumulate values and perform a final computation.
      */
     private inner class Accumulator(
-        var current: Number? = 0L,
-        val nextFunc: (Number?, ExprValue) -> Number,
+        var current: ExprValue?,
```

Comment:
The only thing that makes me a little nervous about this change is that we don't really have a lot of tests surrounding aggregates in general.... Nothing like what you're adding below for the other functions.  Please consider adding additional tests.

Reply:
In particular, with `sum` we need to ensure there are overflow (and underflow) conditions being tested in legacy and permissive modes.

Reply:
Also, can you give some examples related to overflow and underflow?

Reply:
> Can we create a new issue to add more test cases?

I would try to put them in this PR if I could since you can't be certain you're not introducing some bug.

> Also, can you give some examples related to overflow and underflow?

This would be when the result of summation exceeds the maximum value that can be represented with an 8 byte value.  i.e. `SUM([1, 9223372036854775807])`.  What should happen?  Should it throw an exception, or should it coerce into a decimal which can grow to larger values?  (I'm not entirely certain myself since I haven't given this much thought.)  Underflow is the opposite: i.e. `SUM([-1, -9223372036854775808])`  In any case, I looked and I don't think we have any tests for this behavior today, and you should make sure this change doesn't alter it.

Reply:
I tried `SUM([1, 9223372036854775807])` for both the original code and this revision. Both of them gave: 
```
org.partiql.lang.eval.EvaluationException: Int overflow or underflow
	Evaluator Error: at line 1, column 1: INT-8 overflow or underflow
```